gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/InstantiatingHttpJsonChannelProvider.java

```diff
@@ -119,11 +120,22 @@ public boolean needsEndpoint() {
     return endpoint == null;
   }
 
+  @Override
+  public boolean needsEndpointContext() {
+    return false;
+  }
+
   @Override
   public TransportChannelProvider withEndpoint(String endpoint) {
     return toBuilder().setEndpoint(endpoint).build();
   }
 
+  @Override
+  public TransportChannelProvider withEndpointContext(EndpointContext endpointContext) {
+    throw new UnsupportedOperationException(
+        "InstantiatingHttpJsonChannelProvider doesn't need an endpointContext");
+  }
```

Comment:
I think HttpJson should store the valid endpointContext. It'll be used in the future when mTLS is supported.

Reply:
Done in https://github.com/googleapis/sdk-platform-java/commit/393190a1ad03838ff5a3e13222c75e349f0be72e